File: nim/src/autd3/gain.nim
# File: gain.nim
# Project: autd3
# Created Date: 13/06/2022
# Author: Shun Suzuki
# -----
# Last Modified: 13/06/2022
# Modified By: Shun Suzuki (suzuki@hapis.k.u-tokyo.ac.jp)
# -----
# Copyright (c) 2022 Shun Suzuki. All rights reserved.
#


import body
import native_methods/autd3capi

type Gain* = object of Body

proc `=destroy`(gain: var Gain) =
    if (gain.p != nil):
        AUTDDeleteGain(gain.p)
        gain.p = pointer(nil)

File: matlab/autd3/Gain.m
%{
%File: Gain.m
%Project: autd3-matlab
%Created Date: 07/06/2022
%Author: Shun Suzuki
%-----
%Last Modified: 07/06/2022
%Modified By: Shun Suzuki (suzuki@hapis.k.u-tokyo.ac.jp)
%-----
%Copyright (c) 2022 Shun Suzuki. All rights reserved.
%
%}

classdef Gain < Body

    methods

        function obj = Gain()
            obj = obj@Body();
        end

        function delete(obj)

            if obj.ptr.Value ~= 0
                calllib('autd3capi', 'AUTDDeleteGain', obj.ptr);
                obj.ptr = libpointer('voidPtr', 0);
            end

        end

    end

end
